Sources/SourceKitLSP/Swift/CodeActions/PackageManifestEdits.swift

```diff
@@ -0,0 +1,166 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import PackageModel
+import PackageModelSyntax
+import SwiftRefactor
+import SwiftSyntax
+
+/// Syntactic code action provider to provide refactoring actions that
+/// edit a package manifest.
+struct PackageManifestEdits: SyntaxCodeActionProvider {
+  static func codeActions(in scope: SyntaxCodeActionScope) -> [CodeAction] {
+    guard
+      let token = scope.firstToken,
+      let call = token.findEnclosingCall()
+    else {
+      return []
+    }
+
+    var actions = [CodeAction]()
+
+    // If there's a target name, offer to create a test target derived from it.
+    if let targetName = call.findStringArgument(label: "name") {
```

Comment:
Should we also check that we are indeed at a `target`? For example I think we shouldn’t offer this when invoking the refactoring on the call to the `Package` initializer or on a `testTarget`.

Reply:
Yeah, I want to validate that we're on a kind of target we know how to build a test for.

Reply:
Done in https://github.com/apple/sourcekit-lsp/pull/1193/commits/4c1e119197f5997af6d9cc6e6f5a404d09e4e470